mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -38,7 +39,13 @@ where
     /// Create a new instance.
     ///
     /// TODO: consider adding some validation of the bucket.
-    pub fn new(bucket_name: &str, client: Client, source_description: &str, block_size: u64) -> Self {
+    pub fn new(
+        bucket_name: &str,
+        client: Client,
+        source_description: &str,
+        block_size: u64,
+        max_object_size: usize,
```

Comment:
This is for limiting the cache only for object up to a max size, right? Could it be a separate change?